cloudquery/sdk/serve/plugin.py

```diff
@@ -133,13 +162,221 @@ def run(self, args):
             help="network to serve on. can be tcp or unix",
         )
 
-        parsed_args = parser.parse_args(args)
+    def _register_package_command(self, subparsers):
+        package_parser = subparsers.add_parser(
+            "package", help="Package the plugin as a Docker image"
+        )
+        package_parser.add_argument(
+            "version", help="version to tag the Docker image with"
+        )
+        package_parser.add_argument("plugin-directory")
+        package_parser.add_argument(
+            "--log-format",
+            type=str,
+            default="text",
+            choices=["text", "json"],
+            help="logging format",
+        )
+        package_parser.add_argument(
+            "--log-level",
+            type=str,
+            default="info",
+            choices=["trace", "debug", "info", "warn", "error"],
+            help="log level",
+        )
+        package_parser.add_argument(
+            "-D",
+            "--dist-dir",
+            type=str,
+            help="dist directory to output the built plugin. (default: <plugin_directory>/dist)",
+        )
+        package_parser.add_argument(
+            "--docs-dir",
+            type=str,
+            help="docs directory containing markdown files to copy to the dist directory. (default: <plugin_directory>/docs)",
+        )
+        package_parser.add_argument(
+            "-m",
+            "--message",
+            type=str,
+            required=True,
+            help="message that summarizes what is new or changed in this version. Use @<file> to read from file. Supports markdown.",
+        )
 
-        if parsed_args.command == "serve":
-            self._serve(parsed_args)
-        else:
-            parser.print_help()
-            sys.exit(1)
+    def _package(self, args):
+        logger = get_logger(args)
+        self._plugin.set_logger(logger)
+        if self._plugin.name() == None or self._plugin.name() == "":
+            raise Exception("plugin name is required")
+        if self._plugin.team() == None or self._plugin.team() == "":
+            raise Exception("plugin team name is required")
+        if self._plugin.kind() == None or self._plugin.kind() == "":
+            raise Exception("plugin kind is required")
+        if self._plugin.dockerfile() == None or self._plugin.dockerfile() == "":
+            raise Exception("plugin dockerfile is required")
+        if (
+            self._plugin.build_targets() == None
+            or len(self._plugin.build_targets()) == 0
+        ):
```

Comment:
Could we change the default values for all the `_plugin.*` attributes so they are never `None`? That would make all these `if` checks a bit cleaner. Or maybe you could add a helper function like:

```python
def _is_empty(val):
    return val == None or val == "" or val == []
```

Reply:
Name, Team and Kind must be set by the author so they can't have defaults.
Dockerfile and build targets we have defaults, but this check is in case a plugin author sets them explicitly to an empty value

Reply:
Added `_is_empty` in https://github.com/cloudquery/plugin-sdk-python/pull/84/commits/a21a6eded60bc40bed4bf7ea80243e9f7f90b2d1

Reply:
I meant doing `dockerfile: str = ""` instead of `dockerfile: str = None`. But `_is_empty` also works nicely.